tests/test_butler.py

```diff
@@ -619,6 +619,13 @@ def setUp(self):
         uri = ButlerURI(config[".datastore.datastore.root"])
         self.bucketName = uri.netloc
 
+        # set up some fake credentials if they do not exist
+        self.usedDummyCredentials = False
```

Comment:
Please move this code to s3utils and do something like:
```python
self.usedDummyCredentials = setTestCredentials()
```
then in `tearDown`:
```python
if self.usedDummyCredentials = clearTestCredentials()
```
where `clearTestCredentials` and `setTestCredentials` are in s3utils. Granted these are only used for tests so you could have a local `s3testUtils.py` in the `tests/` directory instead if you prefer.

Reply:
Actually, `clearTestCredentials` should probably be called `clearEnvCredentials` since it ensures that there are no environment variable credentials and isn't specific to testing.

Reply:
Moved setting and unsetting of AWS credentials as environmental variables into s3utils.